src/coreclr/src/vm/gchelpers.cpp

```diff
@@ -311,9 +306,44 @@ inline void LogAlloc(size_t size, MethodTable *pMT, Object* object)
 #endif
 }
 #else
-#define LogAlloc(size, pMT, object)
+#define LogAlloc( object)
 #endif
 
+// signals completion of the object to GC and sends events if necessary
+template <class TObj>
+void PublishObjectAndNotify(TObj* &orObject, GC_ALLOC_FLAGS flags)
+{
+    _ASSERTE(orObject->HasEmptySyncBlockInfo());
+
+    if (flags & (GC_ALLOC_LARGE_OBJECT_HEAP | GC_ALLOC_PINNED_OBJECT_HEAP))
```

Comment:
>if (flags & (GC_ALLOC_LARGE_OBJECT_HEAP | GC_ALLOC_PINNED_OBJECT_HEAP)) [](start = 4, length = 71)

I would define a flag GC_ALLOC_USER_OLD_HEAP that is (GC_ALLOC_LARGE_OBJECT_HEAP | GC_ALLOC_PINNED_OBJECT_HEAP) so if we ever add another heap there you don't have to change every place you are calling SetMethodTableForUOHObject